### PR TITLE
Fix for #2248

### PR DESCRIPTION
**Pull request by rynowak**
Split into 2 parts for your convenience and enjoyment!

### CODE REVIEW

File: test/WebSites/RoutingWebSite/RoutingWebSite.xproj [hunk 1]
```diff
@@ -10,6 +10,12 @@
     <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'=='' ">..\..\..\artifacts\obj\$(MSBuildProjectName)</BaseIntermediateOutputPath>
     <OutputPath Condition="'$(OutputPath)'=='' ">..\..\..\artifacts\bin\$(MSBuildProjectName)\</OutputPath>
   </PropertyGroup>
+  <PropertyGroup Label="Configuration" Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <AssemblyName>RoutingWebSite</AssemblyName>
+  </PropertyGroup>
+  <PropertyGroup Label="Configuration">
+    <RootNamespace>RoutingWebSite</RootNamespace>
+  </PropertyGroup>
```
Comment (dougbu) [inline test/WebSites/RoutingWebSite/RoutingWebSite.xproj]: these .xproj properties are useless. don't bother unless VS makes these changes repeatedly.

File: src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultControllerModelBuilder.cs [hunk 2]
```diff
@@ -76,10 +76,57 @@ public ControllerModel BuildControllerModel([NotNull] TypeInfo typeInfo)
...
+                    filteredAttributes.Add(attribute);
+                }
+            }
+            filteredAttributes.AddRange(routeAttributes);
+
+            attributes = filteredAttributes.ToArray();
+
             var controllerModel = new ControllerModel(typeInfo, attributes);
+            AddRange(
+                controllerModel.AttributeRoutes, routeAttributes.Select(a => new AttributeRouteModel(a)));
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultControllerModelBuilder.cs]: ick, YOLO wrapping

File: test/WebSites/RoutingWebSite/RoutingWebSite.xproj (hunk 1 above)
Comment (rynowak) [inline test/WebSites/RoutingWebSite/RoutingWebSite.xproj]: VS did this. I don't think I've ever edited an xproj on purpose.

**Comment by dougbu** [comment]
Change looks good for unnamed routes. But what about uniquely-named routes e.g.
``` c#
  [Route("A", Name="Inherited")]
  public MyController
  {
  }

  [Route("B")]
  public DerivedController : MyController
  {
  }
```
Should the route named "Inherited" be available for the derived controller?

If that decision has already been made, :shipit:

**Comment by rynowak** [comment]
@dougbu - Your intuition is right, we're no longer supporting that on purpose. This feature very quickly becomes a pit of failure when you're try to leverage the 'inherit + add' semantics (removed by this change).

In the example that you give, assume that there's one action in each controller. You'd have a route `A` defined for both of them. You could use the name `Inherited` to generate a link - which would give you a 500 when you visit it....

If you really want to do something like this, you could still easily do it in a convention, which would give much more control.

**Pull request closed without merging**